File: Lab00/Lab00.py
'''
CPE 203 - Lab00
Convert this py file to a java file!
'''
def main():

   # declaring and initializing some variables

   x = 12

   y = 'WELCOME to CSC/CPE203'

   z = 3.1415

   # printing the variables

   print('x:', x, 'y:', y, 'z:', z)

   # a list (make an array in java)

   nums = [5, 16, -8, 2, 11]

   for num in nums:

      print(num)


   # call a function

   numFound = char_count(y, 'C')
   
   print("Number of C in ", y)

   print("Found:", numFound)

   # a counting for loop

   for i in range(1,15):

      print(i, end=' ')

   print()
   
#function counts the given character in the given string
#str str -> int
def char_count(s:str, c:str)-> int:

   count = 0

   for ch in s:

      if ch == c:

         count += 1

   return count


if __name__ == "__main__":

   main()

File: Lab00/src/Lab00.java
/* this part is as comment similar to comments in Python
CPE/CSC 203 Lab00
Name: Noah Giboney

Section: CSC-203-15-2238

*/
public class Lab00 {
   public static void main(String[] arguments) {

      // declaring and initializing some variables

      int x = 12;

      String y = "WELCOME to CSC/CPE203";

      double z = 3.1415;

      // printing the variables

      System.out.println("x: " + x + " " + "y: " + y + " " + "Z: " + z);

      // array in java

      int[] nums = {5,16,-8,2,11};

      for(int i : nums){
         System.out.print(i);
         System.out.println();
      }

      // call a function

      int numFound = charCount(y, 'C');
      System.out.println("Number of C in " + y);
      System.out.println("Found: " + numFound);

      // a counting for loop

      for (int i = 1; i <= 15; i++) {
         System.out.print(i + " ");
      }
      System.out.println();
   }

   public static int charCount(String message, char c) {
      int counter = 0;
      char[] array = message.toCharArray();
      for (int i : array){
         if (i == c){
            counter++;
         }
      }
      return counter;
   }
}